lib/datadog/appsec/configuration/settings.rb

```diff
@@ -126,37 +126,45 @@ def initialize
         end
 
         def enabled
-          @options[:enabled]
+          # Cast for Steep
+          _ = @options[:enabled]
```

Comment:
@lloeki is there no cleaner option for `_ = ...`? I ask because this assignment is not actually needed, and quite distracting.
I dream of a `@options[:enabled]` # steep:cast` or similar comment.

Reply:
Sadly no: types appear to be assigned to each language construct + each value in a way that "pass through" many things, except `_`. So `@options` (and `@options[:enable]`) have consistent types throughout the codebase: there's no form of locality to typing.

Therefore the only way to cast is to go through an assignment to have a second type in there, which can normally go on the left hand side only to a type encompassing the one on the right hand side. To move from a non-encompassing type it needs to go through the special-typed `_` on the left hand side of the assignment, which "swallows" the type information. The new type is then "forced back" by either assignment or return type.

Even the following annotation does not work:

```
        def enabled
          # @type var _cast: untyped
          _cast = @options[:enabled]
        end
```

Indeed the `::Datadog::AppSec::Configuration::DSL::option` type still carries through:

```
lib/datadog/appsec/configuration/settings.rb:128:12: [error] Cannot allow method body have type `::Datadog::AppSec::Configuration::DSL::option` because declared as type `bool`
│   ::Datadog::AppSec::Configuration::DSL::option <: bool
│     ::Datadog::AppSec::Configuration::DSL::option <: (true | false)
│       (::Integer | bool | ::Symbol | ::String | ::StringIO | ::File | ::Regexp | ::Hash[(::Symbol | ::String), ::Datadog::AppSec::Configuration::DSL::option] | ::Array[::Datadog::AppSec::Configuration::DSL::option]) <: (true | false)
│         ::Integer <: (true | false)
│           ::Integer <: true
│
│ Diagnostic ID: Ruby::MethodBodyTypeMismatch
│
└         def enabled
              ~~~~~~~
```

As seen above, `@options[:enabled]` is a value of the `@options` hash, which has a very wide type to be able to store all sorts of types for all possible option keys, but we want the return type of each option to be stricter than the general case.

The only way around that is to have a fully typed, non-generic store, e.g a struct or an actual class instead of a dumb hash with dumb values.

Reply:
Oh wow, I'm kinda with Marco, this is kinda distracting.

I don't think it's a blocker, but here's two suggestions:
* Replace the options actual instance variables. We already need quite a lot of boilerplate to access them inside options, so perhaps making them instance variables would reduce the boilerplate and make steep happy as a side-effect
* Remove the `# Cast for steep` (or maybe just leave the first one). It shows up a bit too often, so it really adds a lot of noise. And if someone thinks this is not needed and removes is, they'll quickly get reminded once they push to CI so it's not like they'll go far without learning it anyway :)